dashboard/client/src/Dashboard.svelte

```diff
@@ -3,23 +3,22 @@
   import ClassDetail from './ClassDetail.svelte';
   import Introduction from './Introduction.svelte';
   import SideNavBar from './SideNavBar.svelte';
-
-  export let chosenChallenge;
-  export let chosenClass;
+  import { chosenChallenge, chosenClass } from './routing';
 </script>
 
 <style>
 </style>
 
 <div class="d-flex">
   <div class="flex-grow-0">
-    <SideNavBar class="mb2" bind:chosenChallenge bind:chosenClass />
+    <SideNavBar />
   </div>
   <div class="ps-5 flex-grow-1">
-    {#if chosenChallenge !== undefined}
-      <ChallengeDetail bind:challenge={chosenChallenge} />
-    {:else if chosenClass !== undefined}
-      <ClassDetail bind:curClass={chosenClass} />
+    {#if $chosenChallenge}
+      <ChallengeDetail challenge={$chosenChallenge} />
+    {:else if $chosenClass}
+      <ClassDetail />
+      <ClassDetail curClass={$chosenClass} />
```

Comment:
Maybe a duplicate line?

Reply:
ah yes, the result of artificially splitting this into multiple commits >:)